x/group/migrations/v2/gen_state.go

```diff
@@ -0,0 +1,55 @@
+package v2
+
+import (
+	"encoding/binary"
+
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+)
+
+// MigrateGenState accepts exported v0.46 x/auth genesis state and migrates it to
+// v0.47 x/auth genesis state. The migration includes:
+// - If the group module is enabled, replace group policy accounts from module accounts to base accounts.
+func MigrateGenState(oldState *authtypes.GenesisState) *authtypes.GenesisState {
+	newState := *oldState
+
+	accounts, err := authtypes.UnpackAccounts(newState.Accounts)
+	if err != nil {
+		panic(err)
+	}
+
+	groupPolicyAccountCounter := uint64(0)
+	for i, acc := range accounts {
+		modAcc, ok := acc.(authtypes.ModuleAccountI)
+		if !ok {
+			continue
+		}
+
+		// Replace group policy accounts from module accounts to base accounts.
+		// These accounts were wrongly created and the address was equal to the module name.
+		if modAcc.GetName() == modAcc.GetAddress().String() {
```

Comment:
I had suggested in my earlier review also inverting this condition and an early return, but if you feel more comfortable this way then no biggie.

Reply:
I actually missed that, I see I only did half of it. Updating 👍🏾